src/models/neural_networks/ealstm.py

```diff
@@ -209,10 +209,131 @@ def forward(self, x, pred_month=None, latlons=None, current=None, yearly_aggs=No
 
 
 class EALSTMCell(nn.Module):
+    """See below. Implemented using modules so it can be explained with shap
+    """
+    def __init__(self,
+                 input_size_dyn: int,
+                 input_size_stat: int,
+                 hidden_size: int,
+                 batch_first: bool = True):
+        super().__init__()
+
+        self.input_size_dyn = input_size_dyn
+        self.input_size_stat = input_size_stat
+        self.hidden_size = hidden_size
+        self.batch_first = batch_first
+
+        self.forget_gate_i = nn.Linear(in_features=input_size_dyn,
+                                       out_features=hidden_size, bias=False)
+        self.forget_gate_h = nn.Linear(in_features=hidden_size, out_features=hidden_size,
+                                       bias=True)
+
+        self.update_gate = nn.Sequential(*[
+            nn.Linear(in_features=input_size_stat, out_features=hidden_size),
+            nn.Sigmoid()
+        ])
+
+        self.update_candidates_i = nn.Linear(in_features=input_size_dyn, out_features=hidden_size,
+                                             bias=False)
+        self.update_candidates_h = nn.Linear(in_features=hidden_size, out_features=hidden_size,
+                                             bias=True)
+
+        self.output_gate_i = nn.Linear(in_features=input_size_dyn, out_features=hidden_size,
+                                       bias=False)
+        self.output_gate_h = nn.Linear(in_features=hidden_size, out_features=hidden_size,
+                                       bias=True)
+
+        self.sigmoid = nn.Sigmoid()
+        self.tanh = nn.Tanh()
+
+        self.reset_parameters()
+
+    def reset_parameters(self):
+        self._reset_i(self.forget_gate_i)
```

Comment:
what is `reset_i`?

```python
    def _reset_i(layer):
        nn.init.orthogonal(layer.weight.data)
```

Reply:
it's setting the `i` matrices to random values?

Reply:
It initializes the weights of the neural networks. `reset_i` initializes the weights of layers on the input (`x[t]`) and `reset_h` initializes weights of layer on the hidden vector `h_0`

This serves the same function as `init_weights` in the linear network.